runtime/src/main/java/org/corfudb/runtime/view/stream/IStreamView.java

```diff
@@ -0,0 +1,123 @@
+package org.corfudb.runtime.view.stream;
+
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.protocols.wireprotocol.TokenResponse;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.UUID;
+import java.util.function.Function;
+
+/** This interface represents a view on a stream. A stream is an ordered
+ * set of log entries which can only be appended to and read in sequential
+ * order.
+ *
+ * Created by mwei on 1/5/17.
+ */
+public interface IStreamView extends Iterator<LogData> {
+
+    /** Return the ID of the stream this view is for.
+     * @return  The ID of the stream.
+     */
+    UUID getID();
+
+    /** Reset the state of this stream view, causing the next read to
+     * start from the beginning of this stream.
+     */
+    void reset();
+
+    /** Append an object to the stream, returning the global address
+     * it was written at.
+     * <p>
+     * Optionally, provide a method to be called when an address is acquired,
```

Comment:
That's not why we need to provide acquire at all. What does CAS have to do with anything? You can't compare and swap on a stream, appends are supposed to be immutable. 

The reason acquire is needed is that acquire is not synchronized, and during append, it is possible for another thread to read an appended entry before append returns. This means that the thread which called append may never see its own write if it calls subsequent reads, because other threads may have already read past it.